cupy/cusparse.py

```diff
@@ -1132,6 +1173,10 @@ def dense2csr(x):
         x.data.ptr, m, nnz_per_row.data.ptr, nnz.ctypes.data)
 
     nnz = int(nnz)
+    if _runtime.is_hip:
+        if nnz == 0:
+            raise ValueError('hipSPARSE currently cannot handle '
+                             'sparse matrices with null ptrs')
```

Comment:
Question: What are these null pointers inside sparse arrays and why do they exist?

Reply:
If nnz is 0, the sparse matrix has no nonzero element, so its data and indices arrays are of zero size. Any sensible implementation would then hold a null pointer (because there is no need to malloc). cuSPARSE handles this situation gracefully but not hipSPARSE/rocSPARSE.

Reply:
Thank you for the explanation. That is indeed a bit crazy.

Reply:
This means that the matrices have all their elements set to 0 and hipSPARSE does something weird to store them?

Reply:
Yeah the matrix has a valid shape but all elements are zero. For this particular case (`rocsparse_create_cXX_descr`) I think rocSPARSE fails to consider nnz=0 as a possibility:
https://github.com/ROCmSoftwarePlatform/rocSPARSE/blob/ab238d05e0a217cc77aa3e4d49df5317d03db2ec/library/src/rocsparse_auxiliary.cpp#L1130
and this is a common pattern across the hipSPARSE/rocSPARSe realm...

cc: @amathews-amd

Reply:
AMD internal ticket: https://ontrack-internal.amd.com/browse/SWDEV-294940

Reply:
Great, thanks for help @amathews-amd! Yeah I see that in some rocSPARSE APIs `nnz=0` is checked to provide a fast-return path (such as [this one](https://github.com/ROCmSoftwarePlatform/rocSPARSE/blob/42431dfb60689b2fae50e741a0b87d6e344f6f0c/library/src/conversion/rocsparse_prune_csr2csr.cpp#L317)), which makes perfect sense, but it's not done in all APIs and thus causing problems.